### PR TITLE
Critical fixes

### PR DESCRIPTION
**Pull request by nuevoalex**
Audit report fixes except for the trading vulnerability.

Also updated some constants

### CODE REVIEW

File: source/contracts/reporting/Reporting.sol [hunk 1]
```diff
@@ -22,8 +22,8 @@ library Reporting {
     uint256 private constant MINIMUM_REPORTING_FEE_DIVISOR = 3; // Maximum 33.3~% fees. Note than anything less than a value of 2 here will likely result in bugs such as divide by 0 cases.
 
     // NOTE: We need to maintain this cost to roughly match the gas cost of reporting. This was last updated 10/02/2017
```
Comment (MicahZoltu) [inline source/contracts/reporting/Reporting.sol]: This date should be updated.

File: source/contracts/trading/CancelOrder.sol [hunk 2]
```diff
@@ -56,15 +59,12 @@ contract CancelOrder is CashAutoConverter, Extractable, ReentrancyGuard, ICancel
             if (_type == Order.Types.Bid) {
                 for (uint8 _i = 0; _i < _market.getNumberOfOutcomes(); ++_i) {
                     if (_i != _outcome) {
-                        _market.getShareToken(_i).transfer(_sender, _sharesEscrowed);
+                        _market.getShareToken(_i).trustedCancelOrderTransfer(_market, _sender, _sharesEscrowed);
```
Comment (MicahZoltu) [inline source/contracts/trading/CancelOrder.sol]: I'm assuming `_market` being unvalidated here is on the TODO list for later?  Should I assume the same for all cases of trusted inputs throughout this PR, or should I call them all out to be sure they don't slip through the cracks?

Comment (nuevoalex) [reply source/contracts/trading/CancelOrder.sol]: Yea I'm doing all of that in another PR currently.

Comment (nuevoalex) [reply source/contracts/trading/CancelOrder.sol]: Which i intend to sit on for a bit until we can get auditor feedback on the two options

**Pull request merged**